scipy/optimize/_direct.py

```diff
@@ -102,6 +102,30 @@ def direct(func, bounds, *, args=(), disp=False,
         ``message`` which describes the cause of the termination. See
         `OptimizeResult` for a description of other attributes.
 
+    Notes
+    -----
+
+    DIRECT is a deterministic
+    optimization algorithm capable of minimizing black box function with
+    its variables subject to lower and upper bound constrains by sampling
+    potential solutions in the search space. The algorithm starts with
+    mapping the hyperrectangle (set of all possible values that can be taken
+    by the input variables subject to the bound constraints) n-dimensional
+    unit hypercube. It samples the function at the center of this hypercube
+    and at 2n (n is the number of variables) more points, 2 in each coordinate
+    direction. Using these function values, DIRECT then divides the domain
+    into hyperrectangles, each having exactly one of the sampling points as
+    its center. In each iteration, DIRECT chooses, using the epsilon parameter,
+    by default 1e-4, some of the existing hyperrectangles to be further divided
+    . This division process continues until the maximum iterations or maximum
+    function evaluations allowed are exceeded, or the function value is within
+    the desired percentage error of the global minimum (if known). The improved
+    version of DIRECT algorithm [1]_ is biased towards local search making it
+    effective for functions without too many local minima. This method wraps
+    the C implementation of the original and improved algorithms.
```

Comment:
Suggestion to highlight the origin of the algorithm

Code history: the original Fortran [implementation](http://www4.ncsu.edu/~ctk/SOFTWARE/DIRECTv204.tar.gz) of DIRECT by Jablonsky was converted to c for the nlopt project [nlopt](https://github.com/stevengj/nlopt). This c implementation was then wrapped in Python for scipy .